src/components/card.js

```diff
@@ -1,32 +1,55 @@
 export default class Card {
-  constructor(data, cardSelector, handleCardClick) {
+  constructor(data, cardSelector, handleCardClick,handleDeleteClick, handleLikeClick) {
     this._name = data.name;
     this._link = data.link;
+    this._id = data._id;
+    this._isLiked = data.isLiked;
     this._cardSelector = cardSelector; // cardSelector = "#card-template"
     this._handleCardClick = handleCardClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleLikeClick = handleLikeClick;
+    
   }
 
   _setEventListeners() {
     //".card__like-button"
     this._cardElement
       .querySelector(".card__like-button")
-      .addEventListener("click", () => this._handeLikeIcon());
+      .addEventListener("click", () => this._handleLikeClick(this));
+    
 
     this._cardElement
       .querySelector(".card__delete-button")
-      .addEventListener("click", () => this._handleDeleteCard());
+      .addEventListener("click", () => this._handleDeleteClick(this));
+
+   
+    /* this._cardElement.querySelector(".modal-delete-button").addEventListener("submit", () => this.handleDeleteCard()); */
+
 
     this._cardImageEl.addEventListener("click", () =>
       this._handleCardClick(this._name, this._link)
     );
+
+   
   }
 
-  _handeLikeIcon() {
+  handleLikeIcon() {
+    
     this._cardElement
       .querySelector(".card__like-button")
-      .classList.toggle("card__like-button_active");
+      .classList.toggle("card__like-button_active")
   }
-  _handleDeleteCard() {
+  _toggleLike(){
+
+    if (this._isLiked){
+      this._cardElement.querySelector(".card__like-button").classList.add("card__like-button_active")
```

Comment:
`this._cardElement.querySelector(".card__like-button")`

**Could be improved**

All elements that are found every time you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** not to waste resources on searching them again and again.

`Class fields` (the so called **class variables** with `this`) are accessible inside any method of the class. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form` , `this._likeButton` and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```

Reply:
Thank you for the suggestion! I've implemented these changes.